es-app/src/guis/GuiGamelistOptions.cpp

```diff
@@ -60,13 +77,26 @@ GuiGamelistOptions::GuiGamelistOptions(Window* window, SystemData* system) : Gui
 GuiGamelistOptions::~GuiGamelistOptions()
 {
 	// apply sort
-	FileData* root = getGamelist()->getCursor()->getSystem()->getRootFolder();
-	root->sort(*mListSort->getSelected()); // will also recursively sort children
-
-	// notify that the root folder was sorted
-	getGamelist()->onFileChanged(root, FILE_SORTED);
+	if (!fromPlaceholder) {
+		FileData* root = getGamelist()->getCursor()->getSystem()->getRootFolder();
+		root->sort(*mListSort->getSelected()); // will also recursively sort children
+
+		// notify that the root folder was sorted
+		getGamelist()->onFileChanged(root, FILE_SORTED);
+	} 
+	if (mFiltersChanged) 
+	{
+		ViewController::get()->reloadGameListView(mSystem);
```

Comment:
Oh, sorry, missed this.

I tried this now, and it works for the most part except that it ends up crashing when the list only has a placeholder. I imagine that the getCursor or setCursor methods end up throwing a segmentation fault somewhere.

The performance hit doesn't seem to be tremendous, though.

Given that I am mostly debugging via logs, if you feel this is a dealbreaker, I'm happy to rig this with logs to see where it ends up crashing.

Or, if you have an easy way to troubleshoot it and debug, if you want to let me know where the problem is I'm happy to take a look if the solution isn't obvious to you!

Sorry and thanks for the help here.